pydis_site/apps/api/viewsets/bot/user.py

```diff
@@ -119,3 +137,15 @@ class UserViewSet(BulkCreateModelMixin, ModelViewSet):
 
     serializer_class = UserSerializer
     queryset = User.objects
+
+    @action(detail=True)
+    def metricity_data(self, request: Request, pk: str = None) -> Response:
+        """Request handler for metricity_data endpoint."""
+        user = self.get_object()
+        with connections['metricity'].cursor() as cursor:
+            data = {}
+            cursor.execute("SELECT verified_at FROM users WHERE id = '%s'", [user.id])
+            data["verified_at"], = cursor.fetchone()
+            cursor.execute("SELECT COUNT(*) FROM messages WHERE author_id = '%s'", [user.id])
+            data["total_messages"], = cursor.fetchone()
```

Comment:
I'm not a fan of this way of unpacking the data. `data["verified_at"],` is a very surprising way to do this, in my opinion. My gut instinct was that it had to be a typo, until I realized that `fetchone()` probably returns _two_ things instead of just one, in spite of its crystal clear method name.

An immediate fix for this would just be to do `[0]` instead of the weird comma, but that's also kind of ugly. Maybe there's a beautiful way to do this?

Reply:
It's because it returns a tuple or list with one element. I agree that it doesn't look great. I plan to wrap the database calls in an abstraction with a nicer API, so that might help. I'll let you know when I push it and you can see what you think.